components/server/src/ome/services/sharing/ShareBean.java

```diff
@@ -752,6 +760,35 @@ public void invalidateConnection(long shareId, Experimenter exp) {
         throw new UnsupportedOperationException();
     }
 
+    @RolesAllowed("user")
+    @Transactional(readOnly = false)
+    public void notifyMembersOfShare(long shareId, String subject, String message,
+            boolean html, List<Experimenter> exps) {
+
+        Set<Long> memberIds = new HashSet<Long>();
+        for (final Experimenter e : getAllMembers(shareId)) {
+            memberIds.add(e.getId());
+        }
+
+        Map<Experimenter, String> errors = new HashMap<Experimenter, String>();
+        for (final Experimenter e : exps) {
```

Comment:
So if no experimenters are passed, this is a no-op? Is that intentional? Should it be possible to notify all without having to pre-load?